<img src="https://versatille.com/wp-content/uploads/The-Top-25-Billionaires-in-the-World-Compared.jpg" width="600px">
<h1>Sobre o Conjunto de Dados</h1>
<p><strong>Descrição:</strong> Este conjunto de dados contém estatísticas sobre os bilionários do mundo, incluindo informações sobre seus negócios, indústrias e detalhes pessoais. Ele fornece insights sobre a distribuição de riqueza, setores de negócios e demografia dos bilionários em todo o mundo.</p>

<h2>Principais Características</h2>
<ul>
    <li><strong>rank:</strong> A classificação do bilionário em termos de riqueza.</li>
    <li><strong>finalWorth:</strong> O patrimônio líquido final do bilionário em dólares americanos.</li>
    <li><strong>category:</strong> A categoria ou indústria na qual opera o negócio do bilionário.</li>
    <li><strong>personName:</strong> O nome completo do bilionário.</li>
    <li><strong>age:</strong> A idade do bilionário.</li>
    <li><strong>country:</strong> O país em que o bilionário reside.</li>
    <li><strong>city:</strong> A cidade em que o bilionário reside.</li>
    <li><strong>source:</strong> A fonte da riqueza do bilionário.</li>
    <li><strong>industries:</strong> As indústrias associadas aos interesses comerciais do bilionário.</li>
    <li><strong>countryOfCitizenship:</strong> O país de cidadania do bilionário.</li>
    <li><strong>organization:</strong> O nome da organização ou empresa associada ao bilionário.</li>
    <li><strong>selfMade:</strong> Indica se o bilionário é auto-suficiente (True/False).</li>
    <li><strong>status:</strong> "D" representa bilionários auto-suficientes (Fundadores/Empreendedores) e "U" indica riqueza herdada ou não conquistada.</li>
    <li><strong>gender:</strong> O gênero do bilionário.</li>
    <li><strong>birthDate:</strong> A data de nascimento do bilionário.</li>
    <li><strong>lastName:</strong> O sobrenome do bilionário.</li>
    <li><strong>firstName:</strong> O primeiro nome do bilionário.</li>
    <li><strong>title:</strong> O título ou honraria do bilionário.</li>
    <li><strong>date:</strong> A data de coleta de dados.</li>
    <li><strong>state:</strong> O estado em que o bilionário reside.</li>
    <li><strong>residenceStateRegion:</strong> A região ou estado de residência do bilionário.</li>
    <li><strong>birthYear:</strong> O ano de nascimento do bilionário.</li>
    <li><strong>birthMonth:</strong> O mês de nascimento do bilionário.</li>
    <li><strong>birthDay:</strong> O dia de nascimento do bilionário.</li>
    <li><strong>cpi_country:</strong> Índice de Preços ao Consumidor (CPI) para o país do bilionário.</li>
    <li><strong>cpi_change_country:</strong> Mudança no CPI para o país do bilionário.</li>
    <li><strong>gdp_country:</strong> Produto Interno Bruto (PIB) para o país do bilionário.</li>
    <li><strong>gross_tertiary_education_enrollment:</strong> Matrícula no ensino terciário no país do bilionário.</li>
    <li><strong>gross_primary_education_enrollment_country:</strong> Matrícula no ensino primário no país do bilionário.</li>
    <li><strong>life_expectancy_country:</strong> Expectativa de vida no país do bilionário.</li>
    <li><strong>tax_revenue_country_country:</strong> Receita fiscal no país do bilionário.</li>
    <li><strong>total_tax_rate_country:</strong> Taxa tributária total no país do bilionário.</li>
    <li><strong>population_country:</strong> População do país do bilionário.</li>
    <li><strong>latitude_country:</strong> Coordenada de latitude do país do bilionário.</li>
    <li><strong>longitude_country:</strong> Coordenada de longitude do país do bilionário.</li>
</ul>

<h2>Potenciais Casos de Uso</h2>
<ul>
    <li><strong>Análise da distribuição de riqueza:</strong> Explore a distribuição da riqueza dos bilionários em diferentes indústrias, países e regiões.</li>
    <li><strong>Análise demográfica:</strong> Investigue a demografia de idade, gênero e local de nascimento dos bilionários.</li>
    <li><strong>Riqueza autoconstruída vs. herdada:</strong> Analise a proporção de bilionários auto-suficientes e aqueles que herdaram sua riqueza.</li>
    <li><strong>Indicadores econômicos:</strong> Estude correlações entre a riqueza dos bilionários e indicadores econômicos como PIB, CPI e taxas fiscais.</li>
    <li><strong>Análise geoespacial:</strong> Visualize a distribuição geográfica dos bilionários e de sua riqueza em um mapa.</li>
    <li><strong>Tendências ao longo do tempo:</strong> Acompanhe as mudanças na demografia e riqueza dos bilionários ao longo dos anos.</li>
</ul>

<hr>

# Importando Bibliotecas

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

os.chdir(r'C:\Users\Anwar\Desktop\O Último Código\Lives\Dados')
pd.set_option('display.max_columns', None)

<hr>

# Importando os Dados

In [33]:
df = pd.read_csv('Billionaires Statistics Dataset.csv', parse_dates = ['birthDate', 'date'])

## Breve análise dos dados

In [34]:
df.head()

rank  finalWorth               category                personName   age  \
0     1      211000       Fashion & Retail  Bernard Arnault & family  74.0   
1     2      180000             Automotive                 Elon Musk  51.0   
2     3      114000             Technology                Jeff Bezos  59.0   
3     4      107000             Technology             Larry Ellison  78.0   
4     5      106000  Finance & Investments            Warren Buffett  92.0   

         country    city              source             industries  \
0         France   Paris                LVMH       Fashion & Retail   
1  United States  Austin       Tesla, SpaceX             Automotive   
2  United States  Medina              Amazon             Technology   
3  United States   Lanai              Oracle             Technology   
4  United States   Omaha  Berkshire Hathaway  Finance & Investments   

  countryOfCitizenship                      organization  selfMade status  \
0               France  LVMH Moët Hennessy Louis Vuitton     False      U   
1        United States                             Tesla      True      D   
2        United States                            Amazon      True      D   
3        United States                            Oracle      True      U   
4        United States    Berkshire Hathaway Inc. (Cl A)      True      D   

  gender  birthDate lastName firstName                 title  \
0      M 1949-03-05  Arnault   Bernard      Chairman and CEO   
1      M 1971-06-28     Musk      Elon                   CEO   
2      M 1964-01-12    Bezos      Jeff  Chairman and Founder   
3      M 1944-08-17  Ellison     Larry       CTO and Founder   
4      M 1930-08-30  Buffett    Warren                   CEO   

                 date       state residenceStateRegion  birthYear  birthMonth  \
0 2023-04-04 05:01:00         NaN                  NaN     1949.0         3.0   
1 2023-04-04 05:01:00       Texas                South     1971.0         6.0   
2 2023-04-04 05:01:00  Washington                 West     1964.0         1.0   
3 2023-04-04 05:01:00      Hawaii                 West     1944.0         8.0   
4 2023-04-04 05:01:00    Nebraska              Midwest     1930.0         8.0   

   birthDay  cpi_country  cpi_change_country           gdp_country  \
0       5.0       110.05                 1.1   $2,715,518,274,227    
1      28.0       117.24                 7.5  $21,427,700,000,000    
2      12.0       117.24                 7.5  $21,427,700,000,000    
3      17.0       117.24                 7.5  $21,427,700,000,000    
4      30.0       117.24                 7.5  $21,427,700,000,000    

   gross_tertiary_education_enrollment  \
0                                 65.6   
1                                 88.2   
2                                 88.2   
3                                 88.2   
4                                 88.2   

   gross_primary_education_enrollment_country  life_expectancy_country  \
0                                       102.5                     82.5   
1                                       101.8                     78.5   
2                                       101.8                     78.5   
3                                       101.8                     78.5   
4                                       101.8                     78.5   

   tax_revenue_country_country  total_tax_rate_country  population_country  \
0                         24.2                    60.7          67059887.0   
1                          9.6                    36.6         328239523.0   
2                          9.6                    36.6         328239523.0   
3                          9.6                    36.6         328239523.0   
4                          9.6                    36.6         328239523.0   

   latitude_country  longitude_country  
0         46.227638           2.213749  
1         37.090240         -95.712891  
2         37.090240         -95.712891  
3         37.090240 

In [35]:
print(f"Temos {df.shape[0]} linhas e {df.shape[1]} colunas")

Temos 2640 linhas e 35 colunas


In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2640 entries, 0 to 2639
Data columns (total 35 columns):
 #   Column                                      Non-Null Count  Dtype         
---  ------                                      --------------  -----         
 0   rank                                        2640 non-null   int64         
 1   finalWorth                                  2640 non-null   int64         
 2   category                                    2640 non-null   object        
 3   personName                                  2640 non-null   object        
 4   age                                         2575 non-null   float64       
 5   country                                     2602 non-null   object        
 6   city                                        2568 non-null   object        
 7   source                                      2640 non-null   object        
 8   industries                                  2640 non-null   object        
 9   countryO

In [37]:
df.gdp_country = df.gdp_country.apply(lambda x: int(x.replace('$', '').replace(',', '')) if not x is np.nan else -1)

In [39]:
df.describe().T

count          mean  \
rank                                        2640.0  1.289159e+03   
finalWorth                                  2640.0  4.623788e+03   
age                                         2575.0  6.514019e+01   
birthYear                                   2564.0  1.957183e+03   
birthMonth                                  2564.0  5.740250e+00   
birthDay                                    2564.0  1.209984e+01   
cpi_country                                 2456.0  1.277552e+02   
cpi_change_country                          2456.0  4.364169e+00   
gdp_country                                 2640.0  1.086333e+13   
gross_tertiary_education_enrollment         2458.0  6.722567e+01   
gross_primary_education_enrollment_country  2459.0  1.028585e+02   
life_expectancy_country                     2458.0  7.812282e+01   
tax_revenue_country_country                 2457.0  1.254624e+01   
total_tax_rate_country                      2458.0  4.396334e+01   
population_country                          2476.0  5.102053e+08   
latitude_country                            2476.0  3.490359e+01   
longitude_country                           2476.0  1.258316e+01   

                                                     std           min  \
rank                                        7.396937e+02      1.000000   
finalWorth                                  9.834241e+03   1000.000000   
age                                         1.325810e+01     18.000000   
birthYear                                   1.328252e+01   1921.000000   
birthMonth                                  3.710085e+00      1.000000   
birthDay                                    9.918876e+00      1.000000   
cpi_country                                 2.645295e+01     99.550000   
cpi_change_country                          3.623763e+00     -1.900000   
gdp_country                                 9.685727e+12     -1.000000   
gross_tertiary_education_enrollment         2.134343e+01      4.000000   
gross_primary_education_enrollment_country  4.710977e+00     84.700000   
life_expectancy_country                     3.730099e+00     54.300000   
tax_revenue_country_country                 5.368625e+00      0.100000   
total_tax_rate_country                      1.214530e+01      9.900000   
population_country                          5.542447e+08  38019.000000   
latitude_country                            1.700350e+01    -40.900557   
longitude_country                           8.676299e+01   -106.346771   

                                                     25%           50%  \
rank                                        6.590000e+02  1.312000e+03   
finalWorth                                  1.500000e+03  2.300000e+03   
age                                         5.600000e+01  6.500000e+01   
birthYear                                   1.948000e+03  1.957000e+03   
birthMonth                                  2.000000e+00  6.000000e+00   
birthDay                                    1.000000e+00  1.100000e+01   
cpi_country                                 1.172400e+02  1.172400e+02   
cpi_change_country                          1.700000e+00  2.900000e+00   
gdp_country                                 1.392681e+12  3.845630e+12   
gross_tertiary_education_enrollment         5.060000e+01  6.560000e+01   
gross_primary_education_enrollment_country  1.002000e+02  1.018000e+02   
life_expectancy_country                     7.700000e+01  7.850000e+01   
tax_revenue_country_country                 9.600000e+00  9.600000e+00   
total_tax_rate_country                      3.660000e+01  4.120000e+01   
population_country                          6.683440e+07  3.282395e+08   
latitude_country                            3.586166e+01  3.709024e+01   
longitude_country                          -9.571289e+01  1.045153e+01   

                                                     75%           max  
rank                                        1.905000e+03  2.540000e+03  
finalWor

In [65]:
df.isnull().mean()[df.isnull().mean() > 0].sort_values(ascending = False)

organization                                  0.876894
title                                         0.871591
residenceStateRegion                          0.717045
state                                         0.714773
cpi_country                                   0.069697
cpi_change_country                            0.069697
tax_revenue_country_country                   0.069318
gross_tertiary_education_enrollment           0.068939
total_tax_rate_country                        0.068939
life_expectancy_country                       0.068939
gross_primary_education_enrollment_country    0.068561
longitude_country                             0.062121
latitude_country                              0.062121
population_country                            0.062121
birthDay                                      0.028788
birthMonth                                    0.028788
birthYear                                     0.028788
birthDate                                     0.028788
city      

## ANÁLISE EXPLORATÓRIA DOS DADOS

### Perguntas de negócio

- De todos os bilionários, quantos são CEOs?
- Quais são as 10 indústrias com mais bilionários?
- ?
- ?
- ?
- ?
- ?

### De todos os bilionários, quantos são CEOs?

In [61]:
# Criando uma lista com valroes 1 para 'ceo' e 0 para não 'ceo' ou nulo
contain_ceo = np.array([0 if type(titulo) != str else 1 if 'ceo' in titulo.lower() else 0 for titulo in df.title.values])

In [66]:
porcentagem_de_ceos = contain_ceo.mean()*100

In [69]:
print(f"Temos {porcentagem_de_ceos:.2f}% de CEO's no dataset.")

Temos 3.52% de CEO's no dataset.


Consideramos como sendo cargos de CEO todos os títulos que continham a palavra 'CEO' nele. Entretanto, temos 87.16% dos dados dessa coluna (title) nulos e isso impacta negativamente a nossa análise. 

Lembrando que a porcentagem foi calculada como sendo a média da lista 'contain_ceo'. Considerei como sendo '1', ou seja, "é um CEO", todos os títulos que continham CEO. Todo o resto (valores nulos e títulos que não contêm 'ceo') considerei como 0.

Abaixo, irei calcular a mesma porcentagem, porém excluindo os valores nulos.

In [70]:
contain_ceo_novo = np.array([-1 if type(titulo) != str else 1 if 'ceo' in titulo.lower() else 0 for titulo in df.title.values])

In [73]:
porcentagem_de_ceos_novo = contain_ceo_novo[contain_ceo_novo != -1].mean()*100

In [76]:
print(f"Temos {porcentagem_de_ceos_novo:.2f}% de CEO's no dataset cujo título é não nulo.")

Temos 27.43% de CEO's no dataset cujo título é não nulo.


A partir disso, vemos que para todos os cargos que são não nulos, 27% dos bilionários desempenham o papel de CEO.

### Quais são as 10 indústrias com mais bilionários?

In [92]:
# Código para responder a nossa pergunta
df.groupby('industries').count()[['rank']].\
            reset_index().\
            sort_values(by = 'rank', ascending = False).\
            rename(columns = {'rank': 'quantity'}).\
            reset_index(drop = True).\
            head(10)

industries  quantity
0  Finance & Investments       372
1          Manufacturing       324
2             Technology       314
3       Fashion & Retail       266
4        Food & Beverage       212
5             Healthcare       201
6            Real Estate       193
7            Diversified       187
8                 Energy       100
9  Media & Entertainment        91

A partir da nossa análise e do dataframe acima, podemos concluir quais são as 10 indústrias com um maior número de bilionários.